android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/TiListView.java

```diff
@@ -367,6 +368,23 @@ public void onScroll(AbsListView view, int firstVisibleItem, int visibleItemCoun
 			{
 				_firstVisibleItem = firstVisibleItem;
 				_visibleItemCount = visibleItemCount;
+				
+				if (listView.getChildAt(0) != null){
+					int currentY = listView.getChildAt(0).getTop();
+					String scrollingDir = null;
+					if (currentY < lastY) {
+						scrollingDir = "up";
+					} else if (currentY > lastY) {
+						scrollingDir = "down";
+					}
+					if (scrollingDir != null){
+						KrollDict eventArgs = new KrollDict();
+						eventArgs.put("direction", scrollingDir);
+						eventArgs.put("velocity", (lastY-currentY));
+						fProxy.fireEvent(TiC.EVENT_SCROLLING, eventArgs, false);
```

Comment:
onScrollStateChange is only called when you start or end scrolling, so you won't get any direction changes, you have to lift your finger and start scrolling again to get the new direction. 

But I can change the event so it will fire only if it gets a new direction, that would reduce the events

If you prefer the onScrollStateChange event the direction parameter could be added to the scrollstart/scrollend callback.